redwood-treehouse-guest/src/jsMain/kotlin/app/cash/redwood/treehouse/ProtocolBridgeJs.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.protocol.Change
+import app.cash.redwood.protocol.ChangesSink
+import app.cash.redwood.protocol.ChildrenTag
+import app.cash.redwood.protocol.Event
+import app.cash.redwood.protocol.Id
+import app.cash.redwood.protocol.ModifierElement
+import app.cash.redwood.protocol.PropertyTag
+import app.cash.redwood.protocol.RedwoodVersion
+import app.cash.redwood.protocol.WidgetTag
+import app.cash.redwood.protocol.guest.ProtocolBridge
+import app.cash.redwood.protocol.guest.ProtocolMismatchHandler
+import app.cash.redwood.protocol.guest.ProtocolWidget
+import app.cash.redwood.protocol.guest.ProtocolWidgetChildren
+import app.cash.redwood.protocol.guest.ProtocolWidgetSystemFactory
+import app.cash.redwood.widget.WidgetSystem
+import app.cash.zipline.asDynamicFunction
+import app.cash.zipline.sourceType
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToDynamic
+
+internal actual fun ProtocolBridge(
+  json: Json,
+  hostVersion: RedwoodVersion,
+  widgetSystemFactory: ProtocolWidgetSystemFactory,
+  mismatchHandler: ProtocolMismatchHandler,
+): ProtocolBridge = FastProtocolBridge(json, hostVersion, widgetSystemFactory, mismatchHandler)
+
+@OptIn(RedwoodCodegenApi::class)
+private class FastProtocolBridge(
```

Comment:
I super dislike this name, but I super don't care when `private`!

Reply:
Can fix!